src/main/java/de/dhbw/Main.java

```diff
@@ -46,14 +48,15 @@ public static void runController(){
         Setting setting;
         int counter = 0;
         while (running) {
-            // message independent code:
-            clock.tick(System.currentTimeMillis());
-            videoIn.grabImage(frame);
-
-            markerRecognizer.setFrame(frame);
-            markerRecognizer.detectShapes();
-            shapeProcessor.processShapes(markerRecognizer.getShapes(), frame.width(), frame.height(), frame);
-            positionMarker.updatePositionMarker(shapeProcessor.getPlayfieldInfo(), clock.currentBeat);
+            if (!stopped) {
```

Comment:
what happens if we move all of the code inside of `if (!stopped){...}` below the `if(stopped){continue}` to reduce the necessary if-clauses? 
and does make sense to move the `shapeProcessor.getFrame() != null` into the `if(stopped)`-clause or is there any case where we do not have a frame but otherwise meaningful UI-message-content?

Reply:
Good point, should be simplified now. I have to separate the ifs though because the frame is always null when the frame-getting code is not executed ^^